installer/steps/dependencies.py

```
@@ -648,35 +737,202 @@ def _install_or_update_plugin(
+    removed_anything |= _legacy_context_mode_uninstall_plugin()
+    removed_anything |= _legacy_context_mode_remove_marketplace()
+    removed_anything |= _legacy_context_mode_remove_orphan_hook()
+    _record_outcome(_OUTCOME_REMOVED if removed_anything else _OUTCOME_UNCHANGED)
+    return True
+
+
+def _legacy_context_mode_uninstall_plugin() -> bool:
+    """Uninstall the context-mode plugin if installed. Idempotent. Returns True if removed."""
+    try:
+        result = subprocess.run(
+            ["claude", "plugin", "list", "--json"],
```

Ruff (0.15.13)

[error] Starting a process with a partial executable path

(S607)